### ISSUE TITLE
Webpack compiler warnings "Failed to parse source map"

### ISSUE THREAD

**Issue by longzheng** [configcat/react-sdk#9]
### Describe the bug

After installing the `configcat-react@^1.0.3` into my Create React App project, I noticed I would get compiler warnings when compiling in either dev mode or build mode.

With `npm run start` I would see this output whenever I saved
```
Compiled with warnings.

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Cache.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Cache.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatContext.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatContext.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHOC.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHOC.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHooks.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHooks.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatProvider.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatProvider.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigFetcher.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigFetcher.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/PollingMode.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/PollingMode.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Version.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Version.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/index.ts'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in ./node_modules/configcat-react/lib/esm/Cache.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Cache.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Cache.ts'

WARNING in ./node_modules/configcat-react/lib/esm/ConfigCatContext.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatContext.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatContext.tsx'

WARNING in ./node_modules/configcat-react/lib/esm/ConfigCatHOC.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHOC.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHOC.tsx'

WARNING in ./node_modules/configcat-react/lib/esm/ConfigCatHooks.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHooks.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHooks.tsx'

WARNING in ./node_modules/configcat-react/lib/esm/ConfigCatProvider.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatProvider.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatProvider.tsx'

WARNING in ./node_modules/configcat-react/lib/esm/ConfigFetcher.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigFetcher.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigFetcher.ts'

WARNING in ./node_modules/configcat-react/lib/esm/PollingMode.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/PollingMode.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/PollingMode.ts'

WARNING in ./node_modules/configcat-react/lib/esm/Version.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Version.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Version.ts'

WARNING in ./node_modules/configcat-react/lib/esm/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/index.ts'

webpack compiled with 9 warnings
No issues found.
```

If I run `npm run build` I would see

```
Creating an optimized production build...
Compiled with warnings.

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Cache.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Cache.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatContext.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatContext.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHOC.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHOC.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHooks.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatHooks.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatProvider.tsx' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigCatProvider.tsx'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigFetcher.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/ConfigFetcher.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/PollingMode.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/PollingMode.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Version.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/Version.ts'

Failed to parse source map from '/Users/long/Desktop/my-app/node_modules/configcat-react/src/index.ts' file: Error: ENOENT: no such file or directory, open '/Users/long/Desktop/my-app/node_modules/configcat-react/src/index.ts'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.
```

### To reproduce

1. Set up a Create React App project `npx create-react-app my-app --template typescript`
2. Install package `npm i configcat-react`
3. Add `ConfigCatProvider` to code, for example I edited `App.tsx` to
```ts
import React from "react";
import logo from "./logo.svg";
import "./App.css";
import { ConfigCatProvider } from "configcat-react";

function App() {
  return (
    <ConfigCatProvider sdkKey="#YOUR_SDK_KEY#">
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.tsx</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
      </div>
    </ConfigCatProvider>
  );
}

export default App;
```
4. Run `npm run start` or `npm run build`

### Expected behavior

No warnings when compiling

### Screenshots

https://user-images.githubusercontent.com/484912/204977079-9a3e65bb-b41b-4a52-8842-76ecb53d6f3c.mov

**Comment by laliconfigcat** [comment]
Hello @longzheng ,

Thanks for your report. I'm checking it and I'll get back to you.

Cheers,
Lali

**Comment by laliconfigcat** [comment]
I created a PR to disable sourcemap generation for our package.
https://github.com/configcat/react-sdk/pull/10

**Comment by laliconfigcat** [comment]
I just released v1.0.4. 
@longzheng, could you please check it? I hope it will solve your problem.

Cheers,

**Comment by longzheng** [comment]
> I just released v1.0.4. @longzheng, could you please check it? I hope it will solve your problem.
> 
> Cheers,

Can confirm this is fixed now. Thanks for the prompt update.

**Comment by laliconfigcat** [comment]
Awesome, thanks for checking it.

Cheers,
Lali
